src/Data/Array.purs

```diff
@@ -698,6 +733,32 @@ modifyAtIndices :: forall t a. Foldable t => t Int -> (a -> a) -> Array a -> Arr
 modifyAtIndices is f xs =
   ST.run (STA.withArray (\res -> traverse_ (\i -> STA.modify i f res) is) xs)
 
+-- | Fold a data structure from the left, keeping all intermediate results
+-- | instead of only the final result. Note that the initial value does not
+-- | appear in the result (unlike Haskell's `Prelude.scanl`).
+-- |
+-- | ```
+-- | scanl (+) 0  [1,2,3] = [1,3,6]
+-- | scanl (-) 10 [1,2,3] = [9,7,4]
+-- | ```
+scanl :: forall a b. (b -> a -> b) -> b -> Array a -> Array b
+scanl = scanlImpl
+
+foreign import scanlImpl :: forall a b. (b -> a -> b) -> b -> Array a -> Array b
```

Comment:
Since the FFI implementation is ready to use as-is, we can just import it as `scanl` straight away; I don't think there's any need to alias it.

Reply:
Fixed in latest commits